src/main/scala/io/iohk/ethereum/network/discovery/PeerDiscoveryManager.scala

```diff
@@ -54,15 +73,17 @@ class PeerDiscoveryManager(
       sendMessage(Pong(to, packet.mdc, expirationTimestamp), from)
 
     case DiscoveryListener.MessageReceived(pong: Pong, from, packet) =>
-      val newNodeInfo = DiscoveryNodeInfo.fromNode(Node(packet.nodeId, from))
-
-      if (nodesInfo.size < discoveryConfig.nodesLimit) {
-        nodesInfo += newNodeInfo.node.id -> newNodeInfo
-        sendMessage(FindNode(ByteString(nodeStatusHolder().nodeId), expirationTimestamp), from)
-      } else {
-        val (earliestNode, _) = nodesInfo.minBy { case (_, node) => node.addTimestamp }
-        nodesInfo -= earliestNode
-        nodesInfo += newNodeInfo.node.id -> newNodeInfo
+      pingedNodes.get(packet.nodeId).foreach { newNodeInfo =>
+        val nodeInfoUpdatedTime = newNodeInfo.copy(addTimestamp = clock.millis())
+        pingedNodes -= newNodeInfo.node.id
+        if (nodesInfo.size < discoveryConfig.nodesLimit) {
```

Comment:
(Minor) This logic seems to be quite similar to `updatePingedNodes` .. maybe we can create a function to share the common stuff (check the size, remove the oldest one, etc)